### PR TITLE
Map rotation and consistency across example activites

### PR DESCRIPTION
**Pull request by cafesilencio**
## Description
Addresses map rotation described in ticket #2676. Also strives for consistency in example activities and makes adjustments that were needed when free drive support was added.

### Goal
Brings a consistency across many of the example activities. Other example activities need to be adjusted still but it's too big a task to do all of them at once. This work effort addresses the map rotation issue described in the original ticket and makes many of the UI example activities consistent in their code.

### CODE REVIEW

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt [hunk 1]
```diff
@@ -63,23 +63,23 @@ class BasicNavigationActivity : AppCompatActivity(), OnMapReadyCallback {
         mapView.getMapAsync(this)
 
         val mapboxNavigationOptions = MapboxNavigation.defaultNavigationOptions(
-            this,
-            Utils.getMapboxAccessToken(this)
+                this,
+                Utils.getMapboxAccessToken(this)
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt:67]: NIT Unnecessary formatting changes. Same thing applies 👇

File: examples/src/main/java/com/mapbox/navigation/examples/core/FasterRouteActivity.kt [hunk 2]
```diff
@@ -117,19 +117,19 @@ class FasterRouteActivity : AppCompatActivity(), OnMapReadyCallback {
...
-                Timber.d("FASTER_ROUTE: millisUntilFinished $millisUntilFinished")
-                fasterRouteAcceptProgress.progress =
-                    (MAX_PROGRESS - millisUntilFinished / COUNT_DOWN_INTERVAL).toInt()
-            }
+            object : CountDownTimer(START_TIME_MILLIS, COUNT_DOWN_INTERVAL) {
+                override fun onTick(millisUntilFinished: Long) {
+                    Timber.d("FASTER_ROUTE: millisUntilFinished $millisUntilFinished")
+                    fasterRouteAcceptProgress.progress =
+                            (MAX_PROGRESS - millisUntilFinished / COUNT_DOWN_INTERVAL).toInt()
+                }
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/FasterRouteActivity.kt:125]: Same as above re: unnecessary formatting changes

File: examples/src/main/java/com/mapbox/navigation/examples/core/InstructionViewActivity.kt [hunk 3]
```diff
@@ -64,25 +59,24 @@ import java.io.File
...
+import kotlinx.android.synthetic.main.activity_instruction_view_layout.startNavigation
 import okhttp3.Cache
 import timber.log.Timber
 
 /**
  * To ensure proper functioning of this example make sure your Location is turned on.
  */
 class InstructionViewActivity : AppCompatActivity(), OnMapReadyCallback,
-    FeedbackBottomSheetListener {
+        FeedbackBottomSheetListener {
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/InstructionViewActivity.kt:72]: Same as above re: unnecessary formatting changes

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReRouteActivity.kt [hunk 4]
```diff
@@ -184,31 +167,21 @@ class ReRouteActivity : AppCompatActivity(), OnMapReadyCallback {
...
-                        .build(),
-                    routesReqCallback
+                mapboxNavigation?.requestRoutes(
+                        RouteOptions.builder().applyDefaultParams()
+                                .accessToken(Utils.getMapboxAccessToken(applicationContext))
+                                .coordinates(originLocation.toPoint(), null, latLng.toPoint())
+                                .alternatives(true)
+                                .profile(DirectionsCriteria.PROFILE_DRIVING_TRAFFIC)
+                                .build(),
+                        routesReqCallback
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/ReRouteActivity.kt:184]: Same as above re: unnecessary formatting changes

File: examples/src/main/java/com/mapbox/navigation/examples/core/ReplayActivity.kt [hunk 5]
```diff
@@ -57,18 +58,20 @@ class ReplayActivity : AppCompatActivity(), OnMapReadyCallback {
         mapView.onCreate(savedInstanceState)
 
         val mapboxNavigationOptions = MapboxNavigation.defaultNavigationOptions(
-            this,
-            Utils.getMapboxAccessToken(this)
+                this,
+                Utils.getMapboxAccessToken(this)
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/ReplayActivity.kt:62]: Same as above re: unnecessary formatting changes

Comment (cafesilencio) [reply examples/src/main/java/com/mapbox/navigation/examples/core/ReplayActivity.kt]: made the formatting changes.

File: examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt [hunk 6]
```diff
@@ -218,18 +227,13 @@ class BasicNavigationActivity : AppCompatActivity(), OnMapReadyCallback {
...
         override fun onSessionStateChanged(tripSessionState: TripSessionState) {
             when (tripSessionState) {
                 TripSessionState.STARTED -> {
-                    stopLocationUpdates()
+                    if (!shouldSimulateRoute()) {
+                        stopLocationUpdates()
+                    }
                 }
                 TripSessionState.STOPPED -> {
                     startLocationUpdates()
```
[the remark this answers is not in the document]
Comment (cafesilencio) [reply examples/src/main/java/com/mapbox/navigation/examples/core/BasicNavigationActivity.kt:239]: It didn't seem necessary for the activity to function and I was avoiding adding anything that wasn't necessary.
